RadioButton (#1167)
jpmorganchase/salt-ds: .changeset/wise-candles-exercise.md

---
"@salt-ds/data-grid": patch
"@salt-ds/lab": patch
---

> 

Reviewer: RadioButtonIcon is also exported and has new props, should that be added?

> RadioButton

Reviewer: no longer extends HTMLAttributes<HTMLDivElement>, should that be noted clearly?

Author: Not sure. Haven't seen any changesets mentioning this. @joshwooding any thoughts on this?

Reviewer 2: I haven't checked but does the move from div to label break anything?

Refactor API and cleanup styles
Remove icon prop; icon is not customisable anymore
Add inputProps

> Add error state

Reviewer: ```suggestion
Added `error` prop for error state styling
```

RadioButtonGroup

> Remove icon prop; icon is not customisable anymore
> Remove legend prop; will be implemented by FormField
> Remove radios prop; should be the users' responsibility
> Add direction prop, replaces row prop
> Add labelWrap prop

Reviewer: ```suggestion
Removed `icon` prop; icon is not customisable anymore
Removed `legend` prop; will be implemented by FormField
Removed `radios` prop; should be the users' responsibility to provide the nested RadioButtons as children
Replaced `row` prop with `direction` prop
Added `labelWrap` prop
```

---
"@salt-ds/data-grid": patch
"@salt-ds/lab": patch

> ---

Reviewer 2: We should add the removal of RadioButtonBase

RadioButton

Refactor API and cleanup styles.

> Removed `icon` prop; icon is not customisable anymore.

Reviewer 2: ```suggestion
Removed `icon` prop; icon is not customizable any more.
```

Added `inputProps` prop to be passed to the radio input.
Added `error` prop for error state styling.

RadioButtonGroup

> Removed `icon` prop; icon is not customisable anymore.

Reviewer 2: ```suggestion
Removed `icon` prop; icon is not customizable any more.
```
